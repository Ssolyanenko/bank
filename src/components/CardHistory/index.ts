export * from './History';
export * from './Pagination';
export * from './CardHistory';
