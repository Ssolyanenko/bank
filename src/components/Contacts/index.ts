export * from './Contacts';
export * from './ContactsFooter';
export * from './ContactsInformation';
