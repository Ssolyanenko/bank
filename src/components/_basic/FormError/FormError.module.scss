@import 'src/styles/variables.module.scss';

.error {
  color: $error_3;
  margin: 8px 0 0 14px;
  font-size: 0.875rem;
}

.storybookWrapper {
  width: 300px;
}
