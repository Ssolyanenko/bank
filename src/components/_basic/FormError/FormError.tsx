import { FC, ReactElement } from 'react';
import { ErrorMessage } from 'formik';

import classes from './FormError.module.scss';

interface Props {
  name: string;
  additionalClass?: string;
}

export const FormError: FC<Props> = ({ name, additionalClass = '' }) => (
  <ErrorMessage name={name}>
    {(message): ReactElement => <div className={`${classes.error} ${additionalClass}`}>{message}</div>}
  </ErrorMessage>
);
