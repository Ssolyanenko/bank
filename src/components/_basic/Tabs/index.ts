export * from './Tabs';
export * from './TabTitle';
