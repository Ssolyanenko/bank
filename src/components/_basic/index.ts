export * from './PrimaryButton';
export * from './Button';
export * from './Input';
export * from './InputPassword';
export * from './ModalWrapper';
export * from './ModalConfirmation';
export * from './PhoneCodeTimer';
export * from './ModalBox';
export * from './CloseButton';
export * from './BackButton';
export * from './StepperForm';
export * from './StepperLabel';
export * from './ChatButton';
export * from './BanTimer';
export * from './CardTemplate';
export * from './SecondaryButton';
export * from './generic';
export * from './SelectStyled';
export * from './TextFieldStyled';
export * from './InputStyled';
export * from './CardTemplate';
export * from './Modal';
export * from './SecureWrapper';
export * from './InputRange';
export * from './DropDown';
export * from './RadioButtonsGroup';
export * from './FormError';
export * from './Checkbox';
export * from './Switcher';
export * from './EditButton';
export * from './CardBlock';
export * from './Tabs';
export * from './ModalSuccess';
export * from './DropDownWithCheckbox';
export * from './SelectFormikTemplate';
export * from './Loader';
export * from './DatePicker';
export * from './CustomRoutes';
