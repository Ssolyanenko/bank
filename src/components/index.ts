export * from './LandingPageComponents';
export * from './Forms';
export * from './ArrowsBlock';
export * from './RootModal';
export * from './Slider';
export * from './Map';
export * from './Contacts';
export * from './MainPageComponents';
export * from './ExchangeRates';
export * from './CardProducts';
export * from './CardHistory';
export * from './SetPinSuccessModal';
export * from './MyCardsApplications';
export * from './MyLoans';
export * from './MyCardsApplications';
export * from './InformationBlock';
export * from './StatusTag';
export * from './DropDownWithSearchBranches';
export * from './FormInputBox';
export * from './MyCards';
export * from './ActivateBlockCardModal';
export * from './UserAccount';
export * from './SetPinCodeFormModal';
export * from './CardDetails';
export * from './CardStatement';
export * from './CardStatementButtons';
export * from './Accordion';
export * from './ReceiptButtons';
export * from './TemporaryComponent/TemporaryComponent';
export * from './TransferModal';
export * from './ModalChoosePaymentCard';
export * from './CardInfo';
export * from './ShareReceiptModal/ShareReceiptModal';
export * from './SocialMedia';
export * from './ResetFormsConfirmModal';
export * from './TransactionLimit';
export * from './ChangeTransactionLimit';
export * from './ActivateBlockCard';
export * from './CardManage';
export * from './ChangePINCode';
export * from './ResponseModal';
export * from './TransactionLimit';
