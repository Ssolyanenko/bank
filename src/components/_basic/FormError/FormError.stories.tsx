import { ComponentMeta, ComponentStory } from '@storybook/react';
import { Field, Formik } from 'formik';

import { FormInputBox } from 'components/FormInputBox';
import { FormError } from './FormError';

import classes from './FormError.module.scss';

export default {
  title: 'Basic/Inputs/InputError',
  component: FormError,
  parameters: {
    docs: {
      description: {
        component: 'Field for error message output',
      },
    },
  },
} as ComponentMeta<typeof FormError>;
const getValidate = (values: { input: '' }): { [field: string]: string } => {
  const errors = { input: '' };

  if (!values.input) errors.input = 'The field is required to fill out';

  return errors;
};

const Template: ComponentStory<typeof FormError> = (props) => (
  <Formik initialValues={{ input: '' }} onSubmit={(): void => {}} validate={getValidate}>
    <div className={classes.storybookWrapper}>
      <Field as={FormInputBox} name="input" />
      <FormError {...props} />
    </div>
  </Formik>
);

export const WithFormInputBox = Template.bind({});

WithFormInputBox.args = {
  name: 'input',
};
