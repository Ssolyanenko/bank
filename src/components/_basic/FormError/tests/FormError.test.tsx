import { render, shallow } from 'enzyme';

import { FormError } from 'components/_basic';
import { WithFormInputBox } from 'components/_basic/FormError/FormError.stories';
import { Formik } from 'formik';

describe('<FormError />', (): void => {
  test('should make a snapshot of component', (): void => {
    const wrapper = render(
      <Formik
        initialValues={{ amount: '' }}
        onSubmit={jest.fn()}
        initialErrors={{ amount: 'error' }}
        initialTouched={{ amount: true }}
      >
        <FormError name="amount" />
      </Formik>
    );
    const component = wrapper.get();

    expect(component).toMatchSnapshot();
  });

  test('should render FormError from storybook', (): void => {
    const props = {
      name: 'input',
    };

    const wrapper = shallow(<WithFormInputBox {...props} />);

    expect(wrapper.find(FormError).length).toBe(1);
  });
});
