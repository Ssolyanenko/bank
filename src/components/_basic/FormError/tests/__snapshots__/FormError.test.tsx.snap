// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormError /> should make a snapshot of component 1`] = `
[
  <div
    class="error "
  >
    error
  </div>,
]
`;
